test/client_tests/client_test.dart

```diff
@@ -399,4 +404,106 @@ void main() {
     expect(Http2ClientConnection('localhost', null, channelOptions).authority,
         'myauthority.com');
   });
+
+  test(
+      'padBase64Multiple4 should pad a string that is not a length multiple of 4',
+      () {
+    final str = '111';
+    final padded = padBase64Multiple4(str);
+    expect(padded, '111=');
+  });
+
+  test(
+      'padBase64Multiple4 should return the original string if it is a multiple of 4',
+      () {
+    final str = '1111';
+    final padded = padBase64Multiple4(str);
+    expect(padded, '1111');
+  });
+
+  test('padBase64Multiple4 should handle empty string', () {
+    final str = '';
+    final padded = padBase64Multiple4(str);
+    expect(padded, '');
+  });
+
+  test('padBase64Multiple4 should handle null string', () {
+    final str = null;
+    final padded = padBase64Multiple4(str);
+    expect(padded, '');
+  });
+
+  test(
+      'decodeStatusDetails should decode details into a List<GeneratedMessage> if base64 present',
+      () {
+    final decodedDetails = decodeStatusDetails(
```

Comment:
I would just test these functions as part of the larger functionality instead of testing them individually.

(We could  put `@visibleForTesting` on them and test them - but I think it is just not worth it).

Reply:
I wasn't sure how to test this by calling the overall function since there seems to be some custom test wrapper functions setup to simulate the requests (`runFailureTest`), so I did this as I knew it would test the functionality I added without anything in between it. Do you prefer we do not do it this way?

Reply:
Testing like this sounds good to me as well, I don't have any objection to it. 

My only request would then be then to mark `decodeStatusDetails` with `@visibleForTesting`.